templates/includes/pagination.html

```diff
@@ -0,0 +1,17 @@
+{% if is_paginated %}
+  <ul class="pagination">
+    {% if page_obj.has_previous %}
+      <li class="page-item">
+        <a href="?page= {{ page_obj.previous_page_number }}" class="page-link">prev</a>
```

Comment:
Remove extra space to avoid url formatting errors

```suggestion
        <a href="?page={{ page_obj.previous_page_number }}" class="page-link">prev</a>
```